exporter/awsemfexporter/config.go

```diff
@@ -59,6 +59,9 @@ type Config struct {
 	// MetricDeclarations is the list of rules to be used to set dimensions for exported metrics.
 	MetricDeclarations []*MetricDeclaration `mapstructure:"metric_declarations"`
 
+	// List of string denoting Gauge metric names required for compaction to values and counts
+	GaugeMetricsToCompact []string `mapstructure:"gauge_metrics_to_compact"`
```

Comment:
nit: Id recommend following the pattern of the other fields that start with `Metric` to imply its something to do with modifying the metrics.
A suggestion would be `MetricAsDistribution`.

This exporter treats gauges and sums the same - in fact further below, it converts gauges to sums.. so should we just respect both for this too?